php: improvements & fixes

tinybeachthor opened the pull request:
- use php extensions from `composer.json` and dependencies
- handle inline version aliases (`dev-master as 1.3.1`)
- refactor & cleanup translators
- refactor semver

aszenz commented:
@tinybeachthor 
Cool the build now works on the main branch: https://github.com/aszenz/php-app-nix/tree/main

But when i try it on a simpler composer.json here on https://github.com/aszenz/php-app-nix/tree/dev-constraint-error it fails showing:

`error: Constraint "dev" could not be parsed`

tinybeachthor commented:
@aszenz 
Fixed now, it was trying to do semver match on all version constraints, even on something like `dev-master`. 
I've added a guard to only semver check constraints where it makes sense.
